portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -64,36 +87,24 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       response.getWriter().println("Please enter a valid comment.");
       return;
     } 
-    if (comments.get(name) == null) {
-      ArrayList<String> commentsFromName = new ArrayList<String>();
-      commentsFromName.add(comment);
-      comments.put(name, commentsFromName);
-    } else {
-      comments.get(name).add(comment);
-    }
+
+    // get the current timestamp
```

Comment:
Time and dates are one of the trickiest thing we have to deal with in programming.

Using the local date and time of the server might cause some issues. For example, if someone outside of the server's timezone is looking at your site, the timezone won't match up properly.

[java.time.Instant](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html) provides a timezone-agnostic representation of a given instant in time. If we have an Instant, we can convert to the appropriate timezone when a user fetches comments.

There are two problems we have to solve here though:

1. Datastore [does not support](http://cloud/appengine/docs/standard/java/datastore/entities#Properties_and_value_types) Instant (it does support java.util.Date, but this still have the localization issue).

2. We need to find the best place to convert to a user's timezone when we display the comments.

For [1], we can get around this by converting to a primitive data type with Datastore support: long. Instant.getEpochSecond() will give you a long representation of the number of seconds since the Unix epoch, and Instant.toEpochMilli will give you a long representing the number of milliseconds. Basically the epoch is just an arbitrary instant in time used in programs which is defined to be 0. This lets us not care about what timezone we're in when we store it.

For [2], we need a way to figure out the user's timezone. I did a quick Google search and it looks like the preferred practice is to do this on the client side, using Javascript functions to determine the local timezone.

With the timezone-agnostic timestamp, and a given user's local timezone, you can then display the timestamps in the local time using Javascript's Date (which has a constructor accepting epoch milliseconds, and a toLocaleString() method for displaying it in the browser's timezone).